lib/grape/validations/validators/values_validator.rb

```diff
@@ -85,7 +85,12 @@ def except_message
         end
 
         def required_for_root_scope?
-          @required && @scope.root?
+          return false unless @required
+
+          scope = @scope
+          scope = scope.parent while scope.lateral?
```

Comment:
I've updated the logic here to handle nested parameters within a `with` block. The method now traverses up the scope hierarchy to check if the parameter's root scope is required, ensuring correct validation even when parameters are deeply nested within a `with' block.

Reply:
You could extract this into `lateral_scope` or something like that, although I am not sure it's worth it